scripts/process_modis.py

```diff
@@ -0,0 +1,399 @@
+import xarray as xr
+import numpy as np
+import datetime as datetime
+import pandas as pd
+import os
+
+
+def calculate_global_mean(data, dataset):
+    """
+    Calculate a global mean, weighted by grid area at a given latitude
+    Parameters:
+    - data (np.ndarray): 3D numpy array containing the data values to be averaged,
+    - dataset (xarray.Dataset): An xarray Dataset object that must contain the latitude coordinates.
+      These are used to calculate the weights for the area-weighted mean. The latitude coordinates
+      should be accessible via dataset.lat.
+
+    Returns:
+    - float: Global mean value
+    """
+
+    factor = np.cos(np.deg2rad(dataset.lat))
+    factor.name = "factor"
+
+    # xarray method
+    global_mean = data.weighted(factor).mean(("lon", "lat", "date"))
+
+    # manual method
+    # global_mean = np.nansum(data * factor, axis=(0, 1, 2)) / (
+    #    np.nansum(factor) * data.shape[0] * data.shape[2]
+    # )
+    return global_mean
+
+
+def calculate_monthly_means(dataset, variable, mask):
+    """
+    Calculate monthly global means from a yearly dataset.
+    If mask is True, then interpolate and mask the data (though this is
+    not needed is using the '_masked' datasets.
+    """
+
+    processed_results = {}
+    if dataset["date"].dtype.kind != "M":
+        valid_dates = dataset.date.where(dataset["date"] >= 0)
+        dataset["date"] = (
+            "date",
+            pd.to_datetime(valid_dates.values, unit="s", errors="coerce"),
+        )
+    monthly_data = dataset.groupby("date.month")
+
+    for month, group in monthly_data:
+        if variable == "TIWP":
+            IWP = group.Cloud_Water_Path_Ice_Mean
+            CF_ice = group.Cloud_Retrieval_Fraction_Ice
+            data_filtered = (
+                IWP.where(IWP >= 0).fillna(0) * CF_ice.where(CF_ice >= 0) * 0.0001
+            )
+        elif variable == "CF":
+            CF = group.Cloud_Fraction_Mean
+            data_filtered = CF.where(CF >= 0) * 0.0001
```

Comment:
The scaling factor is only used once because IWP doesn't need to be scaled. From the file spec:
"_Cloud_Water_Path_Ice_Mean:scale_factor = 1.0d ;
 Cloud_Water_Path_Ice_Mean:add_offset = 0.0d ;_"